python/smashbox/multiprocessing_engine.py

```diff
@@ -208,6 +214,8 @@ def run():
         for p in _smash_.all_procs:
             p.join()
 
+        smashbox.utilities.finalize_test()
```

Comment:
No problem to add exit codes. However maybe just throwing an Exception from these functions is good enough? It reports an error and the whole script exits with code 1. If --loop then it breaks the loop unless --keep-going.

Give it a try and see if that's enough for you. Do you need to control the exit code value?